components/google-cloud/google_cloud_pipeline_components/experimental/HyperparameterTuningJob/component.py

```diff
@@ -0,0 +1,213 @@
+                    Optional. The number of worker replicas. If replica count = 1 then one chief
+                    replica will be provisioned. If replica_count > 1 the remainder will be
+                    provisioned as a worker replica pool.
+                machine_type (str):
+                    Optional. The type of machine to use for training.
+                accelerator_type (str):
+                    Optional. Hardware accelerator type. One of ACCELERATOR_TYPE_UNSPECIFIED,
+                    NVIDIA_TESLA_K80, NVIDIA_TESLA_P100, NVIDIA_TESLA_V100, NVIDIA_TESLA_P4,
+                    NVIDIA_TESLA_T4
+                accelerator_count (int):
+                    Optional. The number of accelerators to attach to a worker replica.
+
+    Returns:
+        List of HyperparameterTuningJob trials
+    """
+
+    aiplatform.init(project=project, location=location,
+                staging_bucket=staging_bucket)
+
+    deserializer = utils.get_deserializer(dict)
+    parameter_spec_kwargs = deserializer(parameter_spec)
+
+    custom_job_display_name = display_name + '_custom_job'
+
+    if local_script_kwargs is None:
+        job = aiplatform.CustomJob(
+            display_name=custom_job_display_name,
+            base_output_dir=base_output_dir,
+            worker_pool_specs=worker_pool_specs,
+        )
+    else:
+        job = aiplatform.CustomJob.from_local_script(
+            display_name=custom_job_display_name,
+            script_path=local_script_kwargs.get('script_path'),
+            container_uri=local_script_kwargs.get('container_uri'),
+            args=local_script_kwargs.get('args'),
+            requirements=local_script_kwargs.get('requirements'),
+            environment_variables=local_script_kwargs.get('environment_variables'),
+            replica_count=local_script_kwargs.get('replica_count'),
+            machine_type=local_script_kwargs.get('machine_type'),
+            accelerator_type=local_script_kwargs.get('accelerator_type'),
+            accelerator_count=local_script_kwargs.get('accelerator_count'),
+            base_output_dir=base_output_dir,
+            encryption_spec_key_name=encryption_spec_key_name,
+            staging_bucket=staging_bucket
+        )
+
+    hp_job = aiplatform.HyperparameterTuningJob(
+        display_name=display_name,
+        custom_job=job,
+        metric_spec=metric_spec,
+        parameter_spec={
+            **parameter_spec_kwargs
+        },
+        max_trial_count=max_trial_count,
+        parallel_trial_count=parallel_trial_count,
+        max_failed_trial_count=max_failed_trial_count,
+        search_algorithm=search_algorithm,
+        measurement_selection=measurement_selection,
+        encryption_spec_key_name=encryption_spec_key_name
+    )
+
+    hp_job.run()
+
+    trials = [json.dumps(trial) for trial in hp_job.trials]
+
+    return trials
+
+
+if __name__ == '__main__':
+    hyperparameter_tuning_job_op = create_component_from_func_v2(
```

Comment:
I wonder how would you share this component if this is inside __name__ == '__main__'?